### DISCUSSION TITLE
Lesson 6: Brownie compiling error

### DISCUSSION THREAD

**Discussion by markreni** [smartcontractkit/full-blockchain-solidity-course-py#631] [Q&A]
Hi,

I'm having some problems when compiling FundMe.sol. I'm getting this kind of error message in my terminal:

  File "brownie/_cli/__main__.py", line 64, in main
    importlib.import_module(f"brownie._cli.{cmd}").main()
  File "brownie/_cli/compile.py", line 50, in main
    proj = project.load()
  File "brownie/project/main.py", line 751, in load
    return Project(name, project_path)
  File "brownie/project/main.py", line 183, in __init__
    self.load()
(base) Marks-MBP:brownie_fund_me bobo$ brownie compile
Brownie v1.17.2 - Python development framework for Ethereum

  File "brownie/_cli/__main__.py", line 64, in main
    importlib.import_module(f"brownie._cli.{cmd}").main()
  File "brownie/_cli/compile.py", line 50, in main
    proj = project.load()
  File "brownie/project/main.py", line 751, in load
    return Project(name, project_path)
  File "brownie/project/main.py", line 183, in __init__
    self.load()
  File "brownie/project/main.py", line 238, in load
    self._compile(changed, self._compiler_config, False)
  File "brownie/project/main.py", line 90, in _compile
    _install_dependencies(self._path)
  File "brownie/project/main.py", line 757, in _install_dependencies
    install_package(package_id)
  File "brownie/project/main.py", line 776, in install_package
    if urlparse(package_id).scheme in ("erc1319", "ethpm"):
  File "/Users/bobo/opt/anaconda3/lib/python3.8/urllib/parse.py", line 372, in urlparse
    url, scheme, _coerce_result = _coerce_args(url, scheme)
  File "/Users/bobo/opt/anaconda3/lib/python3.8/urllib/parse.py", line 124, in _coerce_args
    return _decode_args(args) + (_encode_result,)
  File "/Users/bobo/opt/anaconda3/lib/python3.8/urllib/parse.py", line 108, in _decode_args
    return tuple(x.decode(encoding, errors) if x else '' for x in args)
  File "/Users/bobo/opt/anaconda3/lib/python3.8/urllib/parse.py", line 108, in <genexpr>
    return tuple(x.decode(encoding, errors) if x else '' for x in args)
AttributeError: 'dict' object has no attribute 'decode'

Would anybody have any idea what is wrong with compiling?

Thank you in advance,
Mark

**Comment by cromewar** [comment]
Hello @MarkReni  Could you please share the contract code here in order to check it?

**Reply by markreni** [reply]
Hi @cromewar, yes sure thank u!  


```Solidity
// SPDX-License-Identifier: MIT

pragma solidity ^0.6.0 <0.9.0;

import "@chainlink/contracts/src/v0.6/interfaces/AggregatorV3Interface.sol";
import "@chainlink/contracts/src/v0.6/vendor/SafeMathChainlink.sol";

contract FundMe {
    using SafeMathChainlink for uint256;

    mapping(address => uint256) public addressToAmountFunded;
    address[] public funders;
    address public owner;

    constructor() public {
        owner = msg.sender;
    }

    function fund() public payable {
        uint256 minimumUSD = 50 * 10**18;
        require(
            getConversionRate(msg.value) >= minimumUSD,
            "You need to spend more ETH!"
        );
        addressToAmountFunded[msg.sender] += msg.value;
        funders.push(msg.sender);
    }

    function getVersion() public view returns (uint256) {
        AggregatorV3Interface priceFeed = AggregatorV3Interface(
            0x8A753747A1Fa494EC906cE90E9f37563A8AF630e
        );
        return priceFeed.version();
    }

    function getPrice() public view returns (uint256) {
        AggregatorV3Interface priceFeed = AggregatorV3Interface(
            0x8A753747A1Fa494EC906cE90E9f37563A8AF630e
        );
        (, int256 answer, , , ) = priceFeed.latestRoundData();
        return uint256(answer * 10000000000);
    }

    // 1000000000
    function getConversionRate(uint256 ethAmount)
        public
        view
        returns (uint256)
    {
        uint256 ethPrice = getPrice();
        uint256 ethAmountInUsd = (ethPrice * ethAmount) / 1000000000000000000;
        return ethAmountInUsd;
    }

    modifier onlyOwner() {
        require(msg.sender == owner);
        _;
    }

    function withdraw() public payable onlyOwner {
        msg.sender.transfer(address(this).balance);

        for (
            uint256 funderIndex = 0;
            funderIndex < funders.length;
            funderIndex++
        ) {
            address funder = funders[funderIndex];
            addressToAmountFunded[funder] = 0;
        }
        funders = new address[](0);
    }
}
```

**Reply by cromewar** [reply]
on withdraw function `funderIndex++` you have missing a `;`, also please don't use anaconda for web3 development it's not full compatible yet, use plain python with pip instead, or WSL. additionally don't forget to install ganache-cli, as It seem is not running.
